File: 2020/1/solve_p2.py
#!/usr/bin/env python3

from itertools import product


def main():
    xs = open('in.txt', 'r').readlines()
    xs = list(map(int, xs))

    solution = next(a * b * c
                    for ((a, b), c) in product(product(xs, xs), xs)
                    if a + b + c == 2020)

    print(solution)


if __name__ == '__main__':
    main()

File: 2020/1/solve_p2.hs
#!/usr/bin/env runhaskell

main :: IO ()
main = do
  content <-  readFile "in.txt"

  let nums = map (\x -> read x :: Int) $ lines content
      solution = head [x * y * z | x <- nums, y <- nums, z <- nums, x + y + z == 2020]

  print solution
